tests/Kernel/MediaRenderTest.php

```diff
@@ -59,44 +61,101 @@ protected function setUp() {
+        // File wrapper.
+        $file_wrapper = $crawler->filter('.ecl-file');
+        $this->assertCount(1, $file_wrapper);
+
+        // File row.
+        $file_row = $crawler->filter('.ecl-file .ecl-file__container');
+        $this->assertCount(1, $file_row);
+
+        $file_title = $file_row->filter('.ecl-file__title');
+        $this->assertContains('test document', $file_title->text());
+
+        $file_info_language = $file_row->filter('.ecl-file__info div.ecl-file__language');
+        $this->assertContains($document_language->getName(), $file_info_language->text());
+
+        $file_info_properties = $file_row->filter('.ecl-file__info div.ecl-file__meta');
+        $this->assertContains('KB - PDF)', $file_info_properties->text());
+
+        $file_download_link = $file_row->filter('.ecl-file__download');
+        $this->assertContains('/test_' . $document_langcode . '.pdf', $file_download_link->attr('href'));
+        $this->assertContains('Download', $file_download_link->text());
+
+        // Check the rendering of the available translations.
+        // File translation list.
+        $translation_list = $crawler->filter('.ecl-file__translation-list');
+        $this->assertCount(1, $translation_list);
+
+        // Check the translation list description.
+        $translation_list_description = $translation_list->filter('.ecl-file__translation-item.ecl-file__translation-description');
+        $this->assertContains('Looking for another language which is not on the list? Find out why.', $translation_list_description->text());
+
+        // Get the available translation language.
+        $translation_language = array_filter($translation_languages, function ($langcode) use ($document_langcode) {
+          return $document_langcode != $langcode;
+        }, ARRAY_FILTER_USE_KEY);
+        /** @var \Drupal\Core\Language\Language $translation_language */
+        $translation_language = reset($translation_language);
+
+        $language_names = $predefined = EuropeanUnionLanguages::getLanguageList() + LanguageManager::getStandardLanguageList();
+        $translation_file_info_language = $translation_list->filter('.ecl-file__translation-item .ecl-file__translation-info div.ecl-file__translation-title');
+        $this->assertContains($language_names[$translation_language->getId()][1], $translation_file_info_language->text());
+
+        $translation_file_info_properties = $translation_list->filter('.ecl-file__translation-item .ecl-file__translation-info div.ecl-file__translation-meta');
+        $this->assertContains('KB - PDF)', $translation_file_info_properties->text());
+
+        $translation_file_download_link = $translation_list->filter('.ecl-file__translation-download');
+        $this->assertContains('/test_' . $translation_language->getId() . '.pdf', $translation_file_download_link->attr('href'));
+        $this->assertContains('Download', $file_download_link->text());
+      }
```

Comment:
Test also for a document without translations.

Reply:
Good point.